### PR TITLE
Typescript support

### PR DESCRIPTION
**Pull request by nicodh**
Kickoff to migrate to typescript step by step
TODO (incomplete):
- [x] create useful type definitions
- [x] create useful tslint config
- [ ] check if deployment routine needs changes

fell free to add more points here...

### CODE REVIEW

**Comment by Simon-Laux** [comment]
when you're already tinkering with webpack, you could try if https://parceljs.org/ can give us a performance boost or not.

**Comment by nicodh** [comment]
I would like to merge this soon since it would result in merge conflicts with most of future changes.
It has a dependency to the master of deltachat-nodes with index.d.ts declaration

**Comment by Simon-Laux** [comment]
Is it done in you opinion? (this still has the WIP Tag)

File: src/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  ipcReady: boolean;
+  localeData?: any
+  state?: any;
+  saveState?: (arg: AppState) => void;
+  translate?: (key: string, substitutions: Array<string>, opts: any) => string;
+}
+
+export interface Credentials {
+  addr: string;
+  passwd?: string;
```
Comment (Simon-Laux) [inline src/types.d.ts]: this should be here should it?

Comment (nicodh) [reply src/types.d.ts]: Yes it should, since sometimes the credentials are passed without password like in deltachat state, so the password is optional.

Comment (Simon-Laux) [reply src/types.d.ts]: and where is it used?

**Comment by Simon-Laux** [comment]
Nonetheless I would put this in project resurrection as its to big a change when we have like 5 open prs next to it open

**Comment by pabzm** [comment]
To be honest, I would also prefer to postpone this change. I see that a lot of work has been put into it, but without meaningful tests I'm afraid that it will break things, which would be unfortunate in the current situation where we head towards v1.0.

**Comment by Simon-Laux** [comment]
Nah it will not break much thing, but it is a big task to convert it all and we can only do it when no other pr are currently open. We should meet a week and convert it all to ts together.

**Comment by nicodh** [comment]
I don't think it will break much but if there are no resources now to review I'm fine with postponing. But I would really avoid to "convert it all to ts" in one big change. That would only result in another huge PR no one can really review. 
I added the first step as single PR which has very few changes: https://github.com/deltachat/deltachat-desktop/pull/1333/files
So it is very unlikely that it will "break things"...

**Comment by Simon-Laux** [comment]
I'll close this for now

**Pull request closed without merging**